### PR TITLE
Raz/mgmt 1583

### PR DESCRIPTION
**Pull request by RazRegev**

### CODE REVIEW

File: internal/host/monitor.go [hunk 1]
```diff
@@ -15,8 +15,18 @@ func (m *Manager) HostMonitoring() {
 		log       = requestid.RequestIDLogger(m.log, requestID)
 	)
 
-	monitorStates := []string{HostStatusDiscovering, HostStatusKnown, HostStatusDisconnected, HostStatusInsufficient,
-		HostStatusPendingForInput, models.HostStatusPreparingForInstallation}
+	monitorStates := []string{
```
Comment (filanov) [inline internal/host/monitor.go:18]: what are the states that we don't monitor?

File: internal/host/validator.go [hunk 2]
```diff
@@ -416,3 +418,24 @@ func (v *validator) printHostnameValid(c *validationContext, status validationSt
...
+	switch status {
+	case ValidationSuccess:
+		return fmt.Sprintf("Cluster %s has no errors", c.cluster.ID.String())
+	case ValidationFailure:
+		return fmt.Sprintf("Cluster %s has errors. Aborting installation of host %s",
+			c.cluster.ID.String(), getRealHostname(c.host, c.inventory))
+	case ValidationPending:
+		return "Missing inventory"
+	default:
+		return fmt.Sprintf("Unexpected status %s", status)
```
Comment (filanov) [inline internal/host/validator.go:439]: Not sure will the user do with this string, do we need it?

**Comment by RazRegev** [comment]
> No unit tests?

why not? added tests to transition_test

File: internal/host/monitor.go (hunk 1 above)
Comment (RazRegev) [reply internal/host/monitor.go:18]: preparing, installing-pending-user-action, resetting-pending-user-action, pending-user-input

File: internal/host/validator.go [hunk 3]
```diff
@@ -420,3 +422,23 @@ func (v *validator) printHostnameValid(c *validationContext, status validationSt
 		return fmt.Sprintf("Unexpected status %s", status)
 	}
 }
+
+func (v *validator) noClusterErrors(c *validationContext) validationStatus {
+	if c.inventory == nil {
```
Comment (ori-amizur) [inline internal/host/validator.go:427]: You don't need the inventory for this validation - so this check is not needed

File: internal/host/validator.go [hunk 4]
```diff
@@ -420,3 +422,23 @@ func (v *validator) printHostnameValid(c *validationContext, status validationSt
...
+	return boolValue(swag.StringValue(c.cluster.Status) != models.ClusterStatusError)
+}
+
+func (v *validator) printNoClusterErrors(c *validationContext, status validationStatus) string {
+	switch status {
+	case ValidationSuccess:
+		return fmt.Sprintf("Cluster %s has no errors", c.cluster.ID.String())
+	case ValidationFailure:
+		return fmt.Sprintf("Cluster %s has errors", c.cluster.ID.String())
+	case ValidationPending:
```
Comment (ori-amizur) [inline internal/host/validator.go:439]: Same as above

File: internal/host/statemachine.go [hunk 5]
```diff
@@ -167,6 +167,19 @@ func NewHostStateMachine(th *transitionHandler) stateswitch.StateMachine {
...
 
+	// Abort host if cluster has errors
+	sm.AddTransition(stateswitch.TransitionRule{
+		TransitionType: TransitionTypeRefresh,
+		SourceStates: []stateswitch.State{
+			stateswitch.State(models.HostStatusInstalling),
+			stateswitch.State(models.HostStatusInstallingInProgress),
+			stateswitch.State(models.HostStatusInstalled),
+		},
+		Condition:        stateswitch.Not(If(NoClusterErrors)),
```
Comment (ori-amizur) [inline internal/host/statemachine.go:178]: 1. Currently there is a Noop transitions for all of these states.  They have to be removed.
2. When there is such transition, you need to do opposite rule (Condition: If(NoClusterErrors)) even if it is without a PostTransition (Noop)

File: internal/host/common.go [hunk 6]
```diff
@@ -23,6 +23,7 @@ const (
 	statusInfoResettingPendingUserAction = "Reboot the host into the installation image to complete resetting the installation"
 	statusInfoPreparingForInstallation   = "Preparing host for installation"
 	statusInfoPreparingTimedOut          = "Cluster is no longer preparing for installation"
+	statusInfoAbortingDueClusterErrors   = "Cluster has been aborted due errors"
```
Comment (ori-amizur) [inline internal/host/common.go:26]: Wrong wording - Suggest "Cluster installation has been aborted due to errors"

File: internal/host/statemachine.go [hunk 7]
```diff
@@ -167,6 +167,33 @@ func NewHostStateMachine(th *transitionHandler) stateswitch.StateMachine {
...
+		PostTransition:   th.PostRefreshHost(statusInfoAbortingDueClusterErrors),
+	})
+
+	// Noop transitions for cluster error
+	for _, state := range []stateswitch.State{
+		stateswitch.State(models.HostStatusInstalling),
+		stateswitch.State(models.HostStatusInstallingInProgress),
+		stateswitch.State(models.HostStatusInstalled),
+	} {
+		sm.AddTransition(stateswitch.TransitionRule{
```
Comment (filanov) [inline internal/host/statemachine.go:189]: Maybe we can add a `NoStateChange` to state object?
Something like 
```
func (sh *stateHost) SetState(state stateswitch.State) error {
	if NoStateChange == state {
		return nil
	}
	sh.host.Status = swag.String(string(state))
	return nil
}
```
@ori-amizur What do you think?

Comment (ori-amizur) [reply internal/host/statemachine.go:189]: Possible, but i think that if statewitch does not find any rule to fire it should not be an error. So, we will not need the noop transitions

Comment (filanov) [reply internal/host/statemachine.go:189]: But it is an error, you expected to do some operation and nothing happened. 
Another option is to ignore specific error https://github.com/filanov/stateswitch/blob/master/statemachine.go#L7

**Pull request merged**